atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractLockAwareTransactionManager.java

```diff
@@ -101,6 +114,22 @@
         return runTaskWithLocksWithRetry(ImmutableList.of(), lockSupplier, task);
     }
 
+    private void refreshAfterLockTimeout(Iterable<HeldLocksToken> lockTokens, TransactionLockTimeoutException ex) {
+        Set<LockRefreshToken> toRequest = ImmutableSet.copyOf(Iterables.transform(lockTokens,
+                new Function<HeldLocksToken, LockRefreshToken>() {
+                    @Nullable
+                    @Override
+                    public LockRefreshToken apply(HeldLocksToken input) {
+                        return input.getLockRefreshToken();
+                    }
+                }));
+        Set<LockRefreshToken> refreshedTokens = getLockService().refreshLockRefreshTokens(toRequest);
+        ImmutableSet<LockRefreshToken> failedTokens = Sets.difference(toRequest, refreshedTokens).immutableCopy();
+        if (!failedTokens.isEmpty()) {
+            throw new TransactionLockTimeoutException("lock tokens did not refresh: " + failedTokens, ex);
+        }
```

Comment:
Can we replace this with:

```
        Set<LockRefreshToken> toRequest = ImmutableSet.copyOf(Iterables.transform(
                lockTokens,
                HeldLocksToken::getLockRefreshToken));
        Set<LockRefreshToken> refreshedTokens = getLockService().refreshLockRefreshTokens(toRequest);
        Set<LockRefreshToken> failedTokens = Sets.difference(toRequest, refreshedTokens);
        if (!failedTokens.isEmpty()) {
            throw new TransactionLockTimeoutException("lock tokens did not refresh: " + failedTokens, ex);
        }
```

(and add a test so people know why this is here)